oop1-construct.js

```diff
@@ -11,96 +32,183 @@ Create a class that can categorize at least 6 of the creatures that you recall o
 
 //your code here...
 
+console.log("Problem 2");
+
+class Creature{
+    constructor(name, type, color, actions){
+        this.name = name;
+        this.type = type;
+        this.color = color;
+        this.actions = actions;
+        }
+        
+    introduce(){
+        console.log(`Hello, I am the ${this.name} from the Amazon! I am a ${this.type}.`);
+        }
+
+    describe(){
+        console.log(`I am ${this.color}!`);
+        }
+
+    doThis(){
+        console.log(`I am super cool because I can ${this.actions.join(', and ')}!`);
+        }
+        
+    }
+
+    const riverDolphin = new Creature('River Dolphin', 'sea mammal', 'pink', ['eat small fish crabs and turtles', 'do tribal magic']);
+    const capybara = new Creature ('Capybara', 'rodent', 'brown', ['eat up to 8 pounds of grass per day', 'live in large groups']);
+    const otter = new Creature ('Giant River Otter', 'mammal', 'brown', ['attack in large groups', 'can attack an anaconda']);
+
+    riverDolphin.doThis();
+    otter.introduce();
+    capybara.describe();
+
 
 
 
 /*3. Convert the following Object Literals into a general class called Shape, and instantiate the class to generate the object literals
 */
-
+ console.log('Problem 3');
 
 class Shape {
     //your code here...
-}
 
-
-const triangle = {
-    name : "triangle",
-    sides : [4,7,7], //lengths of each side
-    base : null,
-    height : null, //determine this height using basic geometry.  You may need to research this one.
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${.5 * this.base * this.height}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(this.base) + (this.sides[1] + this.sides[2]) }`)
+    constructor(name, sides, base, height, length, width, radius){
+        this.name = name;
+        this.sides = sides;
+        this.base = base;
+        this.height = height;
+        this.length = length;
+        this.width = width;
+        this.radius = radius;
+    };
+    calcHeight(){
+        return Math.sqrt(Math.pow(this.sides[1], 2) - Math.pow((.5 * this.base), 2)).toFixed(2);
+    }
+    calcArea(){
+        if (this.name === 'triangle'){
+            console.log(`The area of the ${this.name} is ${0.5 * this.base * this.height}`);
+        } else if (this.name === 'rectangle'){
+            console.log(`The area of the ${this.name} is ${this.length * this.width}`);
+        }else if (this.name === 'circle'){
+            console.log(`The area of the ${this.name} is ${(Math.PI * this.radius ** 2).toFixed(2)}`);
```

Comment:
Not good object oriented design, because this has to be modified each time you add a new shape. calcArea should be overwritten each time you make a new child of Shape class